crates/matrix-sdk/src/widget_api/handler/mod.rs

```diff
@@ -33,12 +38,17 @@ impl<T: Driver> MessageHandler<T> {
                     return Err(Error::WidgetError("Content loaded twice".to_string()));
                 }
 
-                let requested = self.driver.send(SendMeCapabilities).await?;
-                let capabilities = self.driver.initialise(requested).await?;
+                let (req, resp) = Request::new(());
+                self.driver.send(Outgoing::SendMeCapabilities(req)).await?;
+                let options = resp.await.map_err(|_| Error::WidgetDied)?;
+
+                let capabilities = self.driver.initialise(options).await?;
                 self.capabilities = Some(capabilities);
 
-                let approved: CapabilitiesUpdated = self.capabilities.as_ref().unwrap().into();
-                self.driver.send(approved).await?;
+                let approved: CapabilitiesReq = self.capabilities.as_ref().unwrap().into();
+                let (req, resp) = Request::new(approved);
+                self.driver.send(Outgoing::CapabilitiesUpdated(req)).await?;
```

Comment:
Regarding your first question: we can't quite do it as `Request` does not have enough data to construct `Message`. It only has a subset of data that the `Message` needs. That's why the `ToWidget` will then be created by the receiver of the `Outgoing` as it will be aware of the JSON structure of the event and can amend it by additional fields (such as header fields etc).

In other words: `handler` handles the "higher-level" request-response logic of the Widget API. Whereas the `transport` (the future module) will handle the conversion/serialization/deserizalition of the events without really knowing much about the request-reply logic.

> why does this need an await if we already await the response?

Good point. You're probably right and the `await` on the `send()` is not required (is that what you're asking?).